### PR TITLE
APIGW NS JWT Auth

### PR DESCRIPTION
**Pull request by jm96441n**
Changes proposed in this PR:
- Enable NS JWT authn/authz for API Gateway

How I've tested this PR:
Running through configurations in this repo: https://github.com/jm96441n/consul-experiments/tree/main/k8s/jwts

How I expect reviewers to test this PR:
Read the PR

Checklist:
- [X] Tests added
- [x] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: charts/consul/templates/connect-inject-clusterrole.yaml [hunk 1]
```diff
@@ -30,11 +30,14 @@ rules:
   - controlplanerequestlimits
   - routeretryfilters
   - routetimeoutfilters
+  - httprouteauthfilters
```
Comment (jm96441n) [inline charts/consul/templates/connect-inject-clusterrole.yaml:33]: this should be `routeauthfilters`

File: charts/consul/test/unit/crd-routeauthfilters.bats [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bats
+
+load _helpers
+
+@test "httproute-auth-filters/CustomResourceDefinition: enabled by default" {
```
Comment (jm96441n) [inline charts/consul/test/unit/crd-routeauthfilters.bats:5]: these names should probably just be `routeauth-filters`

File: control-plane/api-gateway/binding/result.go [hunk 3]
```diff
@@ -382,59 +386,78 @@ func (l listenerValidationResult) conflictedCondition(generation int64) metav1.C
...
+			errListenerInvalidCertificateRef_NonFIPSRSAKeyLen,
+			errListenerInvalidCertificateRef_FIPSRSAKeyLen:
+			conditions = append(conditions, metav1.Condition{
+				Type:               "ResolvedRefs",
+				Status:             metav1.ConditionFalse,
+				Reason:             "InvalidCertificateRef",
+				ObservedGeneration: generation,
+				Message:            refErr.Error(),
+				LastTransitionTime: now,
+			})
```
Comment (t-eckert) [inline control-plane/api-gateway/binding/result.go:419]: Not for this PR, but something to consider looping back around on: maybe we should have a function that creates `metav1.Condition` objects with the current timestamp given how much of this file is defining those objects. Then we could reduce the amount of code by having a function call that builds the condition.

```go
func newCondition(type string, status metav1.Status, reason string, generation int64, message: string) metav1.Condition {
    return metav1.Condition{
		Type:               type,
		Status:             status,
		Reason:             reason,
		ObservedGeneration: generation,
		Message:            message,
		LastTransitionTime: timeFunc(),
	}
}
```

**Review by t-eckert — APPROVED** [review]
Spectacular work! I'm so happy to see this going into `main`!

**Pull request merged**